Source/JavaScriptCore/wasm/WasmOMGIRGenerator.cpp

```diff
@@ -750,12 +750,82 @@ class OMGIRGenerator {
         return WTFMove(m_exceptionHandlers);
     }
 
+    struct AbstractHeap {
+        enum Kind : uint8_t {
+            Globals,
+            WasmMemory,
+            CallSiteIndex,
+            WasmTables,
+            WasmImports,
+            InstanceFields,
+            TableFields,
+            FuncRefTableFields,
+            FunctionFields,
+            NumKinds
+        };
+
+        Kind kind;
+        unsigned offset : 24;
+
+        inline AbstractHeap(Kind kind_in)
```

Comment:
Does the `inline` specifier do anything here? I thought it was implicit for all member functions.

Reply:
I don't think it does. ALWAYS_INLINE would do something. `inline` also does something when the definition is out of line (It errors if the definition isn't defined in the translation unit rather than a link error).

Reply:
Resolved in latest push, actually some rewrites mean this `Kind` is no longer needed.